docs/src/modules/java/pages/value-entity.adoc

```diff
@@ -55,35 +53,38 @@ include::example$valueentity-counter/src/main/proto/value-entities/counter_api.p
 
 == Implementing behavior
 
-A Value Entity implementation is a Java class annotated with link:{attachmentsdir}/api/com/akkaserverless/javasdk/valueentity/ValueEntity.html[`@ValueEntity` {tab-icon}, window="new"]. The class must have a constructor accepting the entity ID in a parameter annotated with link:{attachmentsdir}/api/com/akkaserverless/javasdk/EntityId.html[`@EntityId` {tab-icon}, window="new"].
-
-As we rely on the code-generation, the class `Counter` gets generated for us, it extends the generated class `CounterInterface` which we're not supposed to change as it gets regenerated in case we update the protobuf descriptors.
+A Value Entity implementation is a Java class where you define how each command is implemented. The class `Counter` gets generated for us based on the `counter_api.proto` and `counter_domain.proto` definitions. Once the `Counter.java` file exist, it is not overwritten, so you can freely add logic to it. `Counter` extends the generated class `AbstractCounter` which we're not supposed to change as it gets regenerated in case we update the protobuf descriptors. AbstractCounter` contains all method signatures corresponding to the API of the service. If you change the API you will see compilation errors in the `Counter` class and have to implement the methods required by `AbstractCounter`.
 
 [source,java]
 .src/main/java/com/example/domain/Counter.java
 ----
 include::java:example$valueentity-counter/src/main/java/com/example/domain/Counter.java[tag=class]
 ----
-<1> The constructor accepts a parameter marked as the `@EntityId` and we keep the value in a field.
+<1> Extends the generated `AbstractCounter`, which extends link:{attachmentsdir}/api/com/akkaserverless/javasdk/valueentity/ValueEntity.html[`ValueEntity` {tab-icon}, window="new"].
+<2> By default, the generated constructor has a `ValueEntityContext` parameter, but you can change this to accept other parameters.
```

Comment:
I think that if we should mention this, we should also mention that you'd then have to inject them when handing off the constructor reference in `Main`, but that would be noisy/long to describe here, so perhaps that should be something detailed in a later section?

Or we don't mention at all and clever users will figure that out...

Reply:
It's mentioned later where Main is described. We could skip it here and only have it in the Main description? Ok to have it at both places, somewhat brief here? I don't think this should forward reference to Main.

Reply:
I'd vote for skip here.

Reply:
done: https://github.com/lightbend/akkaserverless-java-sdk/pull/281/commits/a66a34a030f5e9149ddac7f8c650528dbd1eaca4